### PR TITLE
feat(setup) select preferred resolution

### PR DESCRIPTION
**Pull request by mihhu**

### CODE REVIEW

File: spot-client/src/spot-tv/ui/components/setup/select-media/select-media.js [hunk 1]
```diff
@@ -235,6 +252,13 @@ class SelectMedia extends React.Component {
     _getDefaultDeviceListState() {
         return {
             cameras: [],
+            resolutions: [
+                {
+                    label: this.props.t('setup.noSelection'),
+                    value: ''
+                },
+                ...RESOLUTIONS
```
Comment (saghul) [inline spot-client/src/spot-tv/ui/components/setup/select-media/select-media.js:260]: Please make the labels translatable.

File: spot-client/src/common/utils/meeting.js [hunk 2]
```diff
@@ -129,3 +145,34 @@ export function parseMeetingUrl(url) {
...
+ * @param {string} preferredResolution - The resolution of choice.
+ * @returns {boolean}
+ */
+export const getVideoConstraints = (preferredAspectRatio, preferredResolution) => {
+    const aspectRatio = Number(preferredAspectRatio);
+    const resolution = Number(preferredResolution);
+
+    return {
+        video: {
+            height: {
```
Comment (saghul) [inline spot-client/src/common/utils/meeting.js:162]: You need to add a `aspectRatio: 16 / 9` constraint or 4K won't work.

File: spot-client/src/common/utils/meeting.js [hunk 3]
```diff
@@ -129,3 +145,34 @@ export function parseMeetingUrl(url) {
...
 }
+
+/**
+ * Builds the video constraints based on the preferred aspect ratio and resolution.
+ *
+ * @param {string} preferredAspectRatio - The aspect ratio of choice.
+ * @param {string} preferredResolution - The resolution of choice.
+ * @returns {boolean}
+ */
+export const getVideoConstraints = (preferredAspectRatio, preferredResolution) => {
```
Comment (saghul) [inline spot-client/src/common/utils/meeting.js:156]: Where does the aspect ratio come from? I think we should use 16 / 9 at all times. Regardless of the size of the app window, cameras support stabndard aspect ratios, and requiring 16 / 9 for an in-room device sounds very reasonable.

**Review by saghul — COMMENTED** [review]
Left some comments, PTAL!

File: spot-client/src/common/utils/meeting.js (hunk 3 above)
Comment (mihhu) [reply spot-client/src/common/utils/meeting.js:156]: I used [this](https://github.com/jitsi/jitsi-meet-spot/blob/master/spot-client/src/common/app-state/config/selectors.js#L276), but I'll switch it to 16 / 9.

**Pull request merged**